### PR TITLE
Add support for ILIKE

### PR DESCRIPTION
**Pull request by jhernand**
This patch adds support for the `ILIKE` operation.

### CODE REVIEW

**Comment by jhernand** [comment]
@yaacov please review.

File: pkg/tsl/consts.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ const (
 	RegexOp      = "$regex"
 	NotRegexOp   = "$nregex"
 	LikeOp       = "$like"
+	ILikeOp      = "$ilike"
 	NotLikeOp    = "$nlike"
```
Comment (yaacov) [inline pkg/tsl/consts.go]: can we drop the `NotLikeOp` here and in the walkiers ( for example here https://github.com/yaacov/tree-search-language/blob/master/pkg/walkers/sql/walk.go#L142) ?

Can we generate it now ?

Comment (jhernand) [reply pkg/tsl/consts.go]: Yes; I think it can be dropped. Do you want that in this same pull request? I'd prefer a different one.

Comment (jhernand) [reply pkg/tsl/consts.go]: Regarding your question about generating: I already executed the code generator, if that is what you are asking.

Comment (yaacov) [reply pkg/tsl/consts.go]: > Regarding your question about generating: I already executed the code generator, if that is what you are asking.

No :-)
I asked if we can write a query that will generate it in the tree.

Comment (jhernand) [reply pkg/tsl/consts.go]: No, the parser will never generate that kind of tree; it combines NotOp with Like or Ilike. See #13.

**Comment by yaacov** [comment]
@jhernand 
a - do you need a release ?
b - do you want to take https://github.com/yaacov/tree-search-language/issues/12 , it will require adding testing of "not like/ilike" that we are missing, and taking care of the walkers ( currently the semantic walker does not implement notOp, so that may be tricky  ) ?

**Comment by jhernand** [comment]
Yes, we need a release.

I tried to do it in #13, but I didn't take into consideration the missing notOp implementation. Which walkers don't implement it?

**Comment by yaacov** [comment]
> I tried to do it in #13, but I didn't take into consideration the missing notOp implementation. Which walkers don't implement it?

semantic:
https://github.com/yaacov/tree-search-language/blob/master/pkg/walkers/semantics/walk.go#L225

**Pull request merged**